be/src/vec/functions/function_string.h

```diff
@@ -1038,4 +1041,129 @@ class FunctionStringParseUrl : public IFunction {
     }
 };
 
+template <typename Impl>
+class FunctionMoneyFormat : public IFunction {
+public:
+    static constexpr auto name = "money_format";
+    static FunctionPtr create() { return std::make_shared<FunctionMoneyFormat<Impl>>(); }
+    String get_name() const override { return name; }
+
+    DataTypePtr get_return_type_impl(const DataTypes& arguments) const override {
+        return std::make_shared<DataTypeString>();
+    }
+    DataTypes get_variadic_argument_types_impl() const override {
+        return Impl::get_variadic_argument_types();
+    }
+    size_t get_number_of_arguments() const override { return 1; }
+
+    bool use_default_implementation_for_constants() const override { return true; }
+
+    Status execute_impl(FunctionContext* context, Block& block, const ColumnNumbers& arguments,
+                        size_t result, size_t input_rows_count) override {
+        return Impl::execute_impl(context, block, arguments, result, input_rows_count);
+    }
+};
+
+struct MoneyFormatDoubleImpl {
+    static DataTypes get_variadic_argument_types() { return {std::make_shared<DataTypeFloat64>()}; }
+
+    static Status execute_impl(FunctionContext* context, Block& block,
+                               const ColumnNumbers& arguments, size_t result,
+                               size_t input_rows_count) {
+        auto res_column = ColumnString::create();
+        ColumnPtr argument_column = block.get_by_position(arguments[0]).column;
+
+        auto data_column = assert_cast<const ColumnVector<Float64>*>(argument_column.get());
+        auto result_column = assert_cast<ColumnString*>(res_column.get());
+
+        for (int i = 0; i < input_rows_count; i++) {
+            double value =
+                    MathFunctions::my_double_round(data_column->get_element(i), 2, false, false);
+            StringVal str = StringFunctions::do_money_format(context, fmt::format("{:.2f}", value));
+            result_column->insert_data(reinterpret_cast<const char*>(str.ptr), str.len);
+        }
+
+        block.replace_by_position(result, std::move(res_column));
+        return Status::OK();
+    }
+};
+
+struct MoneyFormatInt64Impl {
+    static DataTypes get_variadic_argument_types() { return {std::make_shared<DataTypeInt64>()}; }
+
+    static Status execute_impl(FunctionContext* context, Block& block,
+                               const ColumnNumbers& arguments, size_t result,
```

Comment:
these execute_impl versions just have a little different.
suggest you to design a common function reducing code lines?

Reply:
> these execute_impl versions just have a little different. suggest you to design a common function reducing code lines?

I do some improve.